pallets/did/src/lib.rs

```diff
@@ -134,26 +174,290 @@ pub mod pallet {
+			authentication: Option<T::AuthenticationAddress>,
+			assertion: Option<T::AssertionAddress>,
+			services: Option<BoundedVec<Service<T>, T::MaxServices>>,
+		) -> DispatchResult {
+			let caller = ensure_signed(origin)?;
+
+			// Try to mutate the document associated with the given DID identifier.
+			Did::<T>::try_mutate(did.clone(), |maybe_doc| -> Result<(), DispatchError> {
+				// If document exists get mutable reference. Otherwise return error
+				let did_doc = maybe_doc.as_mut().ok_or(Error::<T>::DidNotFound)?;
+				// ensure that the caller is the controller of the DID
+				Self::ensure_controller(caller, &did_doc);
+
+				// Try to update the document
+				Self::try_update_did(
+					did_doc,
+					did.clone(),
+					controller,
+					authentication,
+					assertion,
+					services
+				)?;
+
+				// Emit event
+				Self::deposit_event(Event::DidUpdated{
+					did,
+					document: did_doc.clone()
+				});
+				Ok(())
+			})
+		}
+
+		#[pallet::weight(1000000)]
+		pub fn force_update_did(
+			origin: OriginFor<T>,
+			did: DidIdentifierOf<T>,
+			controller:Option<DidIdentifierOf<T>>,
+			authentication: Option<T::AuthenticationAddress>,
+			assertion: Option<T::AssertionAddress>,
+			services: Option<BoundedVec<Service<T>, T::MaxServices>>,
+		) -> DispatchResult {
+			// Origin ONLY GovernanceOrigin
+			T::GovernanceOrigin::ensure_origin(origin)?;
+
+			Did::<T>::try_mutate(did.clone(), |maybe_doc| -> Result<(), DispatchError> {
+				let did_doc = maybe_doc.as_mut().ok_or(Error::<T>::DidNotFound)?;
+
+				Self::try_update_did(
+					did_doc,
+					did.clone(),
+					controller,
+					authentication,
+					assertion,
+					services
+				)?;
+				Self::deposit_event(Event::DidForcedUpdated{
+					did,
+					document: did_doc.clone()
+				});
+				Ok(())
+			})
+		}
+
+		#[pallet::weight(1000000)]
+		pub fn remove_did(
+			origin: OriginFor<T>,
+			did: DidIdentifierOf<T>,
+		) -> DispatchResult {
+			let controller = ensure_signed(origin)?;
+
+			//
+			Did::<T>::try_mutate_exists(did.clone(), |maybe_doc| -> Result<(), DispatchError> {
+				let did_doc = maybe_doc.take().ok_or(Error::<T>::DidNotFound)?;
+				// ensure that the caller is the controller of the DID
+				Self::ensure_controller(controller, &did_doc);
+
+				T::Currency::unreserve(&did.clone().into(), T::DidDeposit::get());
+				Self::deposit_event(Event::DidRemoved{
+					did,
+				});
+				Ok(())
+			})
+		}
+
+		#[pallet::weight(1000000)]
+		pub fn force_remove_did(
+			origin: OriginFor<T>,
+			did: DidIdentifierOf<T>,
+		) -> DispatchResult {
+			// Origin ONLY GovernanceOrigin
+			T::GovernanceOrigin::ensure_origin(origin)?;
+
+			Did::<T>::try_mutate_exists(did.clone(), |maybe_doc| -> Result<(), DispatchError> {
+				let did_doc = maybe_doc.take().ok_or(Error::<T>::DidNotFound)?;
+				T::Currency::unreserve(&did.clone().into(), T::DidDeposit::get());
+				Self::deposit_event(Event::DidForcedRemoved{
+					did,
+				});
+				Ok(())
+			})
```

Comment:
```suggestion
			})?;
			Ok(())
```
Preparing for returning a `DispatchResultWithPostInfo` with another Ok(weight) based on the number of `Services`